packages/features/insights/server/trpc-router.ts

```diff
@@ -37,6 +37,47 @@ const userBelongsToTeamProcedure = authedProcedure.use(async ({ ctx, next, rawIn
     where: membershipWhereConditional,
   });
 
+  // No membership found but teamId was provided
+  // So we can look if team it's an org
+  if (!membership && parse.data.teamId) {
+    // Look if userId is owner/admin of
+    const childTeam = await ctx.prisma.team.findFirst({
+      where: {
+        parentId: ctx.user.organizationId,
+        id: parse.data.teamId,
+      },
+      select: {
+        parentId: true,
+      },
+    });
+    if (!childTeam || !childTeam.parentId) {
+      throw new TRPCError({ code: "UNAUTHORIZED" });
+    }
+
+    // Look if userId is owner/admin of team org
+    const isOwnerAdminOfOrg = await ctx.prisma.membership.findFirst({
+      where: {
+        userId: ctx.user.id,
+        teamId: childTeam.parentId,
+        role: {
+          in: ["OWNER", "ADMIN"],
+        },
+        accepted: true,
+      },
+    });
+    if (isOwnerAdminOfOrg && childTeam) {
+      return next({
+        ctx: {
+          ...ctx,
+          user: {
+            ...ctx.user,
+            isOwnerAdminOfParentTeam: true,
```

Comment:
If no direct membership is found, we then look for parent teams and look if user session is admin/owner.

Reply:
Then we pass down ctx a new prop that user is validated and if isOrg is passed in input we should allow user get child teams data.